dll/win32/msgina/shutdown.c

```diff
@@ -538,6 +561,116 @@ ShutdownDialog(
     return ret;
 }
 
+/*************************************************************************
+ * SetRegistryToBootInNTNativeMode
+ *
+ * NOTES
+ *     Used to append the NT Native shell start-up arguments to the
+ *     BootExecute registry value so that the system will boot into it.
+ */
+VOID SetRegistryToBootInNTNativeMode(VOID)
+{
+    #if 1
+        /* FIXME: Add proper shell arguments once a native shell is included.
+                  Until then, don't pollute the registry.
+        */
+        return;
+    #endif
+
+    LONG ReturnCode;
+    HKEY HiveKey;
+    DWORD RegValueType = 0;
+    DWORD InitialRegValueSize = 0;
+    DWORD FinalRegValueSize = 0;
+    WCHAR* InitialRegValue = NULL;
+    WCHAR* FinalRegValue = NULL;
+    WCHAR NativeShellArguments[] = L"native Hello World!\0";
+    const UINT ShellArgsLength = lstrlen(NativeShellArguments) + 2;
+
+    ReturnCode = RegCreateKeyExW(
+        HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Session Manager", 0, NULL, REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS, NULL, &HiveKey, NULL);
+
+    if (ReturnCode != ERROR_SUCCESS)
+    {
+        TRACE("Unable to open the Session Manager key, error %d", GetLastError());
+        return;
+    }
+
+    ReturnCode = RegQueryValueEx(HiveKey, L"BootExecute", NULL, &RegValueType, NULL, &InitialRegValueSize);
+
+    if (ReturnCode != ERROR_SUCCESS || RegValueType != REG_MULTI_SZ)
+    {
+        TRACE("Unable to grab BootExecute, error %d", GetLastError());
+        return;
+    }
+
+    const UINT InitialRegValueLength = InitialRegValueSize / sizeof(WCHAR);
```

Comment:
I don't think `Length` vs `Size` conveys the difference well. I'd suggest either:
* Use Hungarian notation (`cb` vs `cch` is one of the few cases where that's actually useful) -- "the Win32 way"
* Use a descriptive name, e.g. `InitialRegValueBytes / InitialRegValueChars` or `InitialRegValueByteCount / InitialRegValueCharCount` etc.
* Keep all variables in bytes, divide by `sizeof(WCHAR)` where needed -- "the NT way"

Reply:
Replaced suffixes: Length --> CharCount | Size --> ByteCount